TOKEN_GUIDE.md

```diff
@@ -0,0 +1,132 @@
+
+These headers will allow our API to work well in production and development, but for production you should limit only the origins that should be allowed.
+
+## Input Checks
+To generate the token, we need a few input parameters that we retrieve from the route. We need to check that `channel` and `uid` are not empty. If they are, return a Bad Request Error.
+
+The role determines what type of access the user requesting a token should have. There are two possible roles:
+* publisher - Can publish (or send) their data to the channel. They can also receive data from the channel.
+* subscriber - Can only receive data and cannot publish data.
+
+Those are the only two valid inputs. If one of those inputs is in the URL, we can assign the `role` variable to a number predefined within the `agora-token` package. If the request does not include one of these options, we send back a Bad Request Error.
+
+```ts
+//check for valid channel
+if (!params.channel) {
+    return new Response("channel is required", { status: 400, headers })
+}
+//check for valid role
+let role;
+if (params.role === 'publisher') {
+    role = agoraToken.RtcRole.PUBLISHER;
+} else if (params.role === 'subscriber') {
+    role = agoraToken.RtcRole.SUBSCRIBER
+} else {
+    return new Response("role is incorrect", { status: 400, headers })
+}
+//check for valid uid
+if (!params.uid || params.uid === '') {
+    return new Response("uid is required", { status: 400, headers })
+}
+```
+
+## Generate Token
+At this point, we have all the data needed to generate and return the token. The only values that we haven't defined are the token expiration time and the privilege expiration time. These are in seconds, so I will set them both to 10 minutes (600 seconds).
+
+The privilege expiration time defines how long the user will have their privileges within the channel and access to the channel in general. If the privilege time expires, the user can still listen to the call but can no longer publish. And once the token expires in general, they will lose access to the channel completely.
+
+```ts
+const expireTime = 600;
+const privilegeExpireTime = 600;
+let token;
+
+token = agoraToken.RtcTokenBuilder.buildTokenWithUid(APP_ID, APP_CERTIFICATE, params.channel, params.uid, role, expireTime, privilegeExpireTime);
+
+return new Response(JSON.stringify({
+    rtcToken: token
+}), { headers })
+```
+
+You now have a fully working token server that you can use to develop your applications. You should use this token server primarily for development purposes. If you are building in production, you should add some user verification and other security measures to ensure that the only people joining calls are the ones who are supposed to.
+
+![Token Flow](/assets/token-flow.png)
+
+The conclusion of this experiment is that Astro can handle building backends just as well as content sites and I will keep using it for all my projects.
```

Comment:
This conclusion may be too lean, it feels weak. This would be a good place to allude to the other guides/backend-end services that will expand the project.

Reply:
I agree, I wanted to circle back to the intro by talking about how we stress tested a framework focused on "content-heavy" sites and it passed. Reworded to this:

> Astro passed our stress test with flying colors. It was built to support content-heavy sites but performs just as well for backend projects. Astro is a great solution for full-stack websites. Now that you have a token generator built in Astro, you can add a front end and have a secure video call all from one codebase. Here is a [guide on how to build a Video Call front end with Astro](https://www.agora.io/en/blog/build-a-video-call-app-with-astro-and-reactjs/).